aries_vcx/src/handlers/out_of_band/sender/sender.rs

```diff
@@ -1,53 +1,58 @@
 use crate::handlers::out_of_band::{OutOfBand, GoalCode, HandshakeProtocol};
-use crate::messages::attachment::{AttachmentId, AttachmentEncoding};
+use crate::messages::attachment::AttachmentId;
 use crate::messages::connection::service::ServiceResolvable;
 use crate::messages::a2a::A2AMessage;
 use crate::messages::a2a::message_type::MessageType;
 use crate::messages::a2a::message_family::MessageFamilies;
 use crate::error::prelude::*;
 
-impl OutOfBand {
+#[derive(Default, Debug, PartialEq)]
+pub struct OutOfBandSender {
+    pub oob: OutOfBand
+}
+
+impl OutOfBandSender {
     pub fn create() -> Self {
         Self::default()
     }
 
     pub fn set_label(mut self, label: &str) -> Self {
-        self.label = Some(label.to_string());
+        self.oob.label = Some(label.to_string());
         self
     }
 
-    pub fn set_goal_code(mut self, goal_code: GoalCode) -> Self {
-        self.goal_code = Some(goal_code);
+    pub fn set_goal_code(mut self, goal_code: &GoalCode) -> Self {
+        self.oob.goal_code = Some(goal_code.clone());
         self
     }
 
     pub fn set_goal(mut self, goal: &str) -> Self {
-        self.goal = Some(goal.to_string());
+        self.oob.goal = Some(goal.to_string());
         self
     }
 
-    pub fn append_service(mut self, service: ServiceResolvable) -> Self {
-        self.services.push(service);
-        self
+    pub fn append_service(&mut self, service: &ServiceResolvable) -> VcxResult<()> {
```

Comment:
I think we can just return nothing, or Self? There's no error scenario here

Reply:
I like to keep flexibility to add result-returning methods later without necessary changes rippling through every usage of the method.

Reply:
Uhhh I don't know, by that logic we should return Result everywhere. About the flexibility - I see that as a minus, if you actually add error result into `append_service `, it will compile straight away and you might think you are done, skipping review of how the particular error should be handled. If instead you had to change return type of the `append_service` from `Self` to `Result`, you would also be force to better think about the consequences wherever `append_service ` is called.

Reply:
Resolved :)